jbpm-console-ng-process-runtime/jbpm-console-ng-process-runtime-client/src/main/java/org/jbpm/console/ng/pr/client/editors/instance/details/log/NodeHistoryPopup.java

```diff
@@ -0,0 +1,318 @@
+ * limitations under the License.
+ */
+
+package org.jbpm.console.ng.pr.client.editors.instance.details.log;
+
+import com.github.gwtbootstrap.client.ui.*;
+import com.github.gwtbootstrap.client.ui.DataGrid;
+import com.github.gwtbootstrap.client.ui.SimplePager;
+import com.github.gwtbootstrap.client.ui.constants.ButtonType;
+import com.github.gwtbootstrap.client.ui.constants.ControlGroupType;
+import com.github.gwtbootstrap.client.ui.event.ShownEvent;
+import com.github.gwtbootstrap.client.ui.event.ShownHandler;
+import com.google.gwt.cell.client.Cell;
+import com.google.gwt.cell.client.TextCell;
+import com.google.gwt.core.client.GWT;
+import com.google.gwt.i18n.client.DateTimeFormat;
+import com.google.gwt.safehtml.shared.SafeHtmlBuilder;
+import com.google.gwt.uibinder.client.UiBinder;
+import com.google.gwt.uibinder.client.UiField;
+import com.google.gwt.user.cellview.client.*;
+import com.google.gwt.user.client.ui.FlowPanel;
+import com.google.gwt.user.client.ui.HTMLPanel;
+import com.google.gwt.user.client.ui.Widget;
+import com.google.gwt.view.client.ListDataProvider;
+
+import org.jboss.errai.bus.client.api.messaging.Message;
+import org.jboss.errai.common.client.api.Caller;
+import org.jboss.errai.common.client.api.ErrorCallback;
+import org.jboss.errai.common.client.api.RemoteCallback;
+import org.jbpm.console.ng.ga.model.PortableQueryFilter;
+import org.jbpm.console.ng.ga.model.QueryFilter;
+import org.jbpm.console.ng.ht.model.TaskEventSummary;
+import org.jbpm.console.ng.ht.service.TaskAuditService;
+import org.jbpm.console.ng.pr.client.i18n.Constants;
+import org.jbpm.console.ng.pr.client.util.DataGridUtils;
+import org.uberfire.ext.widgets.common.client.common.popups.BaseModal;
+import org.uberfire.ext.widgets.common.client.common.popups.errors.ErrorPopup;
+import org.uberfire.ext.widgets.common.client.common.popups.footers.GenericModalFooter;
+import org.uberfire.mvp.Command;
+import org.uberfire.paging.PageResponse;
+import org.uberfire.workbench.events.NotificationEvent;
+
+import javax.enterprise.context.Dependent;
+import javax.enterprise.event.Event;
+import javax.inject.Inject;
+
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.Map;
+
+@Dependent
+public class NodeHistoryPopup extends BaseModal {
+
+    interface NodeHistoryPopupBinder extends UiBinder<Widget, NodeHistoryPopup> {
+
+    }
+
+    @UiField
+    public TextBox currentTaskIdTextBox;
+
+    @UiField
+    public TextBox currentTaskNameTextBox;
+
+    @UiField
+    public HelpBlock errorMessages;
+
+    @UiField
+    public ControlGroup errorMessagesGroup;
+
+    @UiField
+    public DataGrid<TaskEventSummary> taskEventListGrid;
+
+    @UiField
+    public FlowPanel listContainer;
+
+    @Inject
+    private Event<NotificationEvent> notification;
+
+    @Inject
+    private Caller<TaskAuditService> taskAuditService;
+
+    private ListDataProvider<TaskEventSummary> dataProvider = new ListDataProvider<TaskEventSummary>();
```

Comment:
We have being working a lot trying to unify the usages of the PagedTable class provided by uberfire-extensions. We should use that in all the places where a data grid is needed.